src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/ECDsaCng.cs

```diff
@@ -75,13 +80,17 @@ private static bool IsEccAlgorithmGroup(CngAlgorithmGroup algorithmGroup)
         private CngAlgorithmCore _core;
 
         // See https://msdn.microsoft.com/en-us/library/windows/desktop/bb931354(v=vs.85).aspx
+        // See https://msdn.microsoft.com/en-us/library/windows/desktop/mt632245%28v=vs.85%29.aspx for named curves
         private static readonly KeySizes[] s_legalKeySizes =
             new KeySizes[]
             {
                 // All values are in bits.
-                new KeySizes(minSize: 256, maxSize: 384, skipSize: 128),
-                new KeySizes(minSize: 521, maxSize: 521, skipSize: 0),
+                new KeySizes(minSize: 160, maxSize: 224, skipSize: 32), // 160, 192, 224
```

Comment:
Take a look at the openssl. I have a todo in there to port that logic to cng. 

I 'froze' the 3 original curves\keysizes, and dynamically added the current curve keysize to the supported keysize array so the base class doesn't throw if it detects an invalid key size. I did ensure the keysize property does work correct for named curves (that is, return the correct key size) even though it isn't all that useful.

I could do what you suggested and allow 'any' key size as a simplification but I prefer the idea of having LegalKeySizes only report the legacy key sizes plus the current key size. Open for input here.